lib.php

```diff
@@ -15,71 +15,99 @@
 // along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
 
 /**
- * Tiny C4L library functions.
+ * Tiny Elements library functions.
  *
- * @package   tiny_c4l
+ * @package   tiny_elements
  * @copyright 2023 Marc Català <reskit@gmail.com>
  * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
-defined('MOODLE_INTERNAL') || die();
+use tiny_elements\local\utils;
 
 /**
- * Return a list of all the user preferences used by tiny_c4l.
+ * Return a list of all the user preferences used by tiny_elements.
  *
  * @return array
  */
-function tiny_c4l_user_preferences() {
+function tiny_elements_user_preferences() {
     $preferences = [];
 
-    $preferences['c4l_components_variants'] = [
-        'type' => PARAM_RAW,
+    $preferences['elements_category'] = [
+        'type' => PARAM_INT,
         'null' => NULL_NOT_ALLOWED,
-        'default' => '',
+        'default' => 0,
+    ];
+
+    $preferences['elements_category_flavors'] = [
+        'type' => PARAM_RAW,
+        'null' => NULL_ALLOWED,
+    ];
+
+    $preferences['elements_component_variants'] = [
+        'type' => PARAM_RAW,
+        'null' => NULL_ALLOWED,
     ];
 
     return $preferences;
 }
 
 /**
- * Serves the tiny_c4l files.
+ * Serve the requested file for the tiny_elements plugin.
  *
- * @param stdClass $course course object
- * @param stdClass $cm course module object
- * @param stdClass $context context object
- * @param string $filearea file area
- * @param array $args extra arguments
+ * @param stdClass $course the course object
+ * @param stdClass $cm the course module object
+ * @param context $context the context
+ * @param string $filearea the name of the file area
+ * @param array $args extra arguments (itemid, path)
  * @param bool $forcedownload whether or not force download
  * @param array $options additional options affecting the file serving
- * @return bool false if file not found, does not return if found - just send the file
+ * @return bool false if the file not found, just send the file otherwise and do not return anything
  */
-function tiny_c4l_pluginfile($course,
+function tiny_elements_pluginfile(
```

Comment:
Do we intentionally send images, js, and css without any capability checks? This means this basically is publicly downloadable. From an admin's perspective it could thought that the components are being kept inside the platform (only for authenticated users for example)

Reply:
Added a check for $CFG->forcelogin